sympy/solvers/tests/test_solveset.py

```diff
@@ -3271,3 +3271,13 @@ def test_issue_22628():
 
 def test_issue_25781():
     assert solve(sqrt(x/2) - x) == [0, S.Half]
+
+
+def test_issue_25423():
+    x, y = symbols('x y')
+    equations1 = []
+    equations2 = [2 * x + 3 * y - 1]
+    nonlinear_equations = [x ** 2 + y ** 2 - 1, x - y]
+    raises(TypeError, lambda: linear_eq_to_matrix(equations1, {x, y}))
+    raises(TypeError, lambda: linear_eq_to_matrix(equations2, {x, y}))
+    raises(NonlinearError, lambda: linear_eq_to_matrix(nonlinear_equations, x, y))
```

Comment:
```suggestion
    raises(ValueError, lambda: linear_eq_to_matrix(equations1, {x, y}))
    raises(ValueError, lambda: linear_eq_to_matrix(equations2, {x, y}))
```
Suggest changing TypeError to ValueError, too, to match behavior when passing equations as a set.

Reply:
This change from TypeError to ValueError will cause test failure becasue within the `linear_eq_to_matrix` function, a _TypeError_ is raised when checking to see if the symbols parameter is a set, indicating that an unordered `set` type is not supported for the symbols parameter. This is the actual behavior of the function checking the `set` input. 

I think since we are concerned with `set` type as input, we should indeed raise _TypeError_ instead of ValueError?

Reply:
Then wouldn't aTypeError be raised for passing a set for symbols, too?